unstructured/ingest/interfaces.py

```diff
@@ -11,12 +12,41 @@
 from unstructured.staging.base import convert_to_dict
 
 
+@dataclass
+class StandardConnectorConfig:
+    """Common set of config options passed to all connectors."""
+
+    # where raw documents are stored for processing, and then removed if not preserve_downloads
+    download_dir: str
+    # where to write structured data outputs
+    output_dir: str
+    download_only: bool = False
+    fields_include: str = "element_id,text,type,metadata"
+    flatten_metadata: bool = False
+    metadata_exclude: Optional[str] = None
+    metadata_include: Optional[str] = None
+    partition_by_api: bool = False
+    partition_endpoint: str = "https://api.unstructured.io/general/v0/general"
+    preserve_downloads: bool = False
+    re_download: bool = False
+
+
+class BaseConnectorConfig(ABC):
+    """Abstract definition on which to define connector-specific attributes."""
+
+
+@dataclass
 class BaseConnector(ABC):
     """Abstract Base Class for a connector to a remote source, e.g. S3 or Google Drive."""
 
-    def __init__(self, config):
-        """Expects a config object that implements BaseConnectorConfig."""
-        pass
+    standard_config: StandardConnectorConfig
+    config: BaseConnectorConfig
+
+    def __init__(self, standard_config: StandardConnectorConfig, config: BaseConnectorConfig):
+        """Expects a standard_config object that implements StandardConnectorConfig
+        and config object that implements BaseConnectorConfig."""
+        self.standard_config = standard_config
+        self.config = config
```

Comment:
I think we could potentially ditch this given this is a dataclass. though maybe helpful to see the definitions here and have a space carved  out if we want to add base init functionality in the future. in the subclasses, adopted the pattern of calling `super().__init__(standard_config, config)` rather than duplicating `self.standard_config = standard_config` and `self.config = config` everywhere, but this also could mean that we can leverage in the future if we need anything to happen on init across all connectors.